src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs

```diff
@@ -1188,6 +1188,11 @@ protected override string GetColumnType(string schema, string table, string name
 
         private static string GetColumnTypeWithCharSetAndCollation(ColumnOperation operation, string columnType)
         {
+            if (columnType.IndexOf("json", StringComparison.OrdinalIgnoreCase) >= 0)
```

Comment:
Please add a support property to `ServerVersionSupport` (e.g. `JsonWithCharsetAndCollation`), override it it in `MariaDbServerVersion.MariaDbServerVersionSupport` and use it in your check here to verify whether charsets and collations are supported for JSON columns or not.

Reply:
Also, add a test that executes the different behavior between MySQL and MariaDB.

Reply:
Actually, my requests are not needed anymore.

_(From https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1365#issuecomment-824461107:)_

> Actually, it turns out that MariaDB seems to force a charset of `utf8mb4` and a collation of `utf8mb4_bin` as well, when using the `json` type (alias for `longtext`), but doesn't if just `longtext` is used.
> So there doesn't need to be a distinction between MySQL and MariaDB here, as both will implicitly use the charset of `utf8mb4` and the `utf8mb4_bin` collation.
> 
> However, since `json` is just an alias for `longtext` for MariaDB, it returns the type `longtext` from `information_schema.columns` when we scaffold it. Because Pomelo will only automatically enable the JSON support for properties with the column type `json`, users would currently need to manually change the as `longtext` scaffolded columns to `json`.
> 
> However, since `10.4.3`, MariaDB will automatically add a `JSON_VALID()` check constraint to `json` columns (see [JSON Data Type](https://mariadb.com/kb/en/json-data-type/) and [MDEV-13916: Enforce check constraint on JSON type](https://jira.mariadb.org/browse/MDEV-13916)), so we can use that to identify a `longtext` column with a charset of `utf8mb4` and a collation of `utf8mb4_bin` as `json`.